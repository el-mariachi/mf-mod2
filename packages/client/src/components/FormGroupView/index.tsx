import Form from 'react-bootstrap/Form'
import { UseFormRegister, FieldErrorsImpl } from 'react-hook-form'

type FormGroupViewProps = {
  formName: string
  controlProps: ControlProps
  register: UseFormRegister<Fields>
  errors: Partial<FieldErrorsImpl>
  readOnly?: boolean
}

type Fields = FormGroupViewProps['controlProps']['name']

export default ({
  controlProps,
  readOnly,
  register,
  errors,
  formName,
}: FormGroupViewProps) => {
  const { name, type, label, placeholder, test, message, validate } =
    controlProps
  const pattern =
    test !== undefined && message !== undefined
      ? {
          value: test,
          message: message,
        }
      : undefined
  return (
    <Form.Group className="mb-3" controlId={`${formName}-${name}`}>
      <Form.Label>{label}</Form.Label>
      <Form.Control
        type={type}
        isInvalid={errors[name] !== undefined}
        placeholder={placeholder}
        disabled={readOnly}
        {...register(name, {
          required: name === 'password' ? false : 'Поле должно быть заполнено',
          pattern,
          validate: name === 'password' ? undefined : validate,
        })}
      />
      <Form.Control.Feedback type="invalid">
        {String(errors[name]?.message)}
      </Form.Control.Feedback>
    </Form.Group>
  )
}
