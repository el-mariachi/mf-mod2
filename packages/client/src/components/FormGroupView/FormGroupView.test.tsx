/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { render } from '@testing-library/react'
import FormGroupView from '.'

describe('#FormGroupView', () => {
  it('SFormGroupView should to match snapshot', () => {
    const registerCallbackStub = () => {}
    const { baseElement } = render(
      <FormGroupView
        formName={'formName'}
        controlProps={{
          name: 'name',
          label: 'label',
          type: 'type',
        }}
        register={registerCallbackStub as any}
        errors={[123] as any}
      />
    )
    expect(baseElement).toMatchSnapshot()
  })
})
