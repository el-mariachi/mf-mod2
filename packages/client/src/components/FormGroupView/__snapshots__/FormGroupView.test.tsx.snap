// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#FormGroupView SFormGroupView should to match snapshot 1`] = `
<body>
  <div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="formName-name"
      >
        label
      </label>
      <input
        class="form-control"
        id="formName-name"
        type="type"
      />
      <div
        class="invalid-feedback"
      >
        undefined
      </div>
    </div>
  </div>
</body>
`;
